package one.clproto.communication_protocol;

message ClusterMsg {
    required string module_name = 1;
    required string message_type = 2;
    required string message_decoder_name = 3;
    required string answer_type = 4;
    required string answer_decoder_name = 5;
    required bool synch = 6;
    required int32 protocol_version = 7;
    optional int32 message_id = 8 [default = 0];
    optional bytes input = 9;
    optional bytes global_user_id = 10;
    optional bytes token_hash = 11;
}

message Answer {
    required string answer_status = 1;
    optional int32 message_id = 2 [default = 0];
    optional string message_type = 3 [default = ""];
    optional bytes worker_answer = 4;
    optional string error_description = 5;
    optional string message_decoder_name = 6;
}

message ProviderMsg {
    required string message_type = 1;
    required bytes input = 2;
    optional bytes fuse_id = 3;
}

message Atom {
    required string value = 1;
}
